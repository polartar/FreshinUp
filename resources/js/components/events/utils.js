import get from 'lodash/get'

const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}

export default getFileNameCopy
