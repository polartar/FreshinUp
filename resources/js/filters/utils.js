export function prefixStr (value, prefix = '@') {
  if (!value) return ''
  return value ? `${prefix} ${value}` : ''
}
