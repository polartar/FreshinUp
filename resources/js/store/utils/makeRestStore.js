/**
 * All of this below is just so that when we call store.dispatch('module/deleteItem', { getItems: true })
 * it will get them with the needed include if that was set previously
 *
 * Example:
 * From
 * store.dispatch('module/getItems', { params: { include: 'submoduleA,submoduleB' } }
 * store.dispatch('module/deleteItem', { getItems: false, id: item.uuid })
 * store.dispatch('module/getItems', { params: { include: 'submoduleA,submoduleB' } }
 *
 * To
 * store.dispatch('module/getItems', { params: { include: 'submoduleA,submoduleB' } }
 * store.dispatch('module/deleteItem', { getItems: true, id: item.uuid })
 *
 * Hey that's one line less and a lot of code duplication avoided for each module
 */

import mapKeys from 'lodash/mapKeys'
import {
  _getPaginationAsParams, _getSortingAsParams,
  buildApi,
  makeModule as _makeModule
} from '@freshinup/core-ui/src/store/utils/makeRestStore'
import pickBy from 'lodash/pickBy'
import omitBy from 'lodash/omitBy'

export const makeModule = (store, moduleName = '') => {
  const m = _makeModule(store, moduleName)
  const _actions = store.actions

  m.mutations = {
    ...m.mutations,
    setFilters: (state, payload) => {
      payload = omitBy(payload, (value, key) => {
        return key.indexOf('sort') > -1
      })
      state.filters = mapKeys(payload, (value, key) => {
        if (['include'].includes(key)) {
          return key
        }
        return key.replace(/filter\[(.*)]/gi, '$1')
      })
    }
  }
  m.actions = {
    ...m.actions,
    getItems (context, payload = { params: {}, data: {} }) {
      const pagination = context.getters.pagination
      const sorting = context.getters.sorting
      // Convert filters into query parameters that is JSON API compliant
      const filters = mapKeys(context.state.filters, function (value, key) {
        // In order to be backwards compatible we need to ignore a few keys
        //    No deprecation plan yet as there is not a design change planned.
        //    Most likely we'll need a setSort() and term may remain
        if (['term', 'q', 'sort', 'include'].indexOf(key) > -1) {
          return key
        }
        return key.indexOf('filter[') === 0 ? key : `filter[${key}]`
      })

      // Deprecation Notice
      const deprecatedKeys = Object.keys(pickBy(payload.params, (value, key) => {
        return key.indexOf('filter[') > -1 || key.indexOf('sort') > -1
      }))
      if (deprecatedKeys.length > 0) {
        console.warn(`
FreshBUS: Version 1.15 of FreshBUS is the last support filtering through getList({ params }) on Store Modules made from makeRestStore.
Please switch to using setFilters() before calling getItems(). Keys provided are ${deprecatedKeys} on ${moduleName}
          `)
      }
      payload.params = {
        ..._getPaginationAsParams(pagination),
        ..._getSortingAsParams(sorting),
        ...filters,
        ...payload.params
      }

      return _actions.getItems.apply(null, [context, payload])
    }
  }
  return m
}

/**
 * Quickly build a REST Module with our Fresh Standards
 * @param name
 * @param items Initial State for Items
 * @param item  Initial State for Item (selected item)
 * @param filters
 * @param basePaths
 * @param options
 */
export default (name, { items, item, filters }, basePaths, options = {}) => {
  const api = buildApi(name, { items, item, filters }, basePaths, options)
  return makeModule(api.getStore(), name)
}
