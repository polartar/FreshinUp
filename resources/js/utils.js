import get from 'lodash/get'
/**
 *
 * @param options HTML attributes
 * @param [options.type] string 'link'|'script' or anything in fact
 * @param [options.container] HTMLElement
 * @returns {Promise<unknown>}
 */
export const lazyLoad = (options) => {
  return new Promise((resolve, reject) => {
    if (!options.type) {
      return reject(new Error('options.type is required'))
    }
    if (!options.container) {
      options.container = document.body
    }
    const { type, container, ...attributes } = options
    let element = document.createElement(type)
    Object.keys(attributes).forEach(key => element.setAttribute(key, attributes[key]))
    element.addEventListener('load', () => {
      resolve(element)
    })
    element.addEventListener('error', (e) => {
      reject(e)
    })
    container.append(element)
  })
}

export const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}
