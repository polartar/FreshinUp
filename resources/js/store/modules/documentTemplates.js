import omit from 'lodash/omit'
import makeRestStore from '../utils/makeRestStore'

export default ({ items, item }) => {
  const store = makeRestStore('foodfleet/document/templates', { items, item })
  store.getters = {
    ...store.getters,
    sortables (state) {
      return [
        { value: '-created_at', text: 'Newest' },
        { value: 'created_at', text: 'Oldest' },
        { value: 'title', text: 'Title (A - Z)' },
        { value: '-title', text: 'Title (Z - A)' }
      ]
    }
  }

  const _createItem = store.actions.createItem
  const _updateItem = store.actions.updateItem
  const unwantedKeys = ['id', 'uuid', 'status', 'updated_by', 'updated_at', 'created_at']
  store.actions = {
    ...store.actions,
    createItem (context, payload) {
      payload.data = omit(payload.data, unwantedKeys)
      return _createItem(context, payload)
    },
    updateItem (context, payload) {
      payload.data = omit(payload.data, unwantedKeys)
      return _updateItem(context, payload)
    }
  }

  store.modules = {
    ...store.modules,
    statuses: makeRestStore('foodfleet/document/template/statuses', { items, item })
  }
  return store
}
