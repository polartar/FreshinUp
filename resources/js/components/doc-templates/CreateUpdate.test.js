import Component from './CreateUpdate'
import { shallowMount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import createStore from 'tests/createStore'
import { mount } from 'vue-cli-plugin-freshinup-ui/utils/testing'
import { FIXTURE_DOCUMENT_TEMPLATES } from '../../../../tests/Javascript/__data__/documentTemplates'

describe('components/doc-templates/CreateUpdate', () => {
  describe('Snapshots', () => {
    let store, mock, localVue
    beforeEach(() => {
      store = createStore()
      let v = createLocalVue()
      localVue = v.localVue
      mock = v.mock
    })
    test('new', (done) => {
      const wrapper = mount(Component, {
        localVue,
        store
      })
      Component.beforeRouteEnterOrUpdate(wrapper.vm, { params: { id: 'new' }}, null, () => {
        expect(wrapper.element).toMatchSnapshot()
        done()
      })
    })
    test('edit', (done) => {
      mock.onGet(/.*foodfleet\/document\/template\/[A-z0-9]+/)
        .reply(200, { data: FIXTURE_DOCUMENT_TEMPLATES })
      const wrapper = mount(Component, {
        localVue,
        store
      })
      Component.beforeRouteEnterOrUpdate(wrapper.vm, { params: { id: FIXTURE_DOCUMENT_TEMPLATES[0].uuid }}, null, () => {
        expect(wrapper.element).toMatchSnapshot()
        done()
      })
    })
  })
  describe('Methods', () => {
    let store, mock, localVue
    beforeEach(() => {
      const v = createLocalVue()
      mock = v.mock
      store = createStore()
    })
    describe('errorVars', () => {
      test('when no variables', async () => {
        const wrapper = shallowMount(Component, {
          localVue,
          store
        })
        const errorVars = wrapper.vm.errorVars(`
          This is a sample text with no variables at all
        `, {})
        expect(errorVars).toHaveLength(0)
      })
      test('when one variable is not valid', async () => {
        const wrapper = shallowMount(Component, {
          localVue,
          store
        })
        const errorVars = wrapper.vm.errorVars(`
          This is a sample text with {{ foo }} as variable at all but not {{ bar }}
        `, {
          foo: 'abc'
        })
        expect(errorVars).toHaveLength(1)
        expect(errorVars).toContain('bar')
      })
      test('when all variables are valid', async () => {
        const wrapper = shallowMount(Component, {
          localVue,
          store
        })
        const errorVars = wrapper.vm.errorVars(`
          This is a sample text with {{ foo }} as variable at all but not {{ bar }}
        `, {
          foo: 'abc',
          bar: 123
        })
        expect(errorVars).toHaveLength(0)
      })
    })
  })
})
