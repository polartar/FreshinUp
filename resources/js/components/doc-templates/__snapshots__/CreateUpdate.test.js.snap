// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/doc-templates/CreateUpdate Snapshots edit 1`] = `
<div>
  <div
    class="layout px-4 pt4 column"
  >
    <div
      class="flex"
    >
      <button
        class="mx-0 v-btn v-btn--flat v-btn--small theme--light"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          <div
            class="d-flex align-content-center white--text"
          >
            <i
              aria-hidden="true"
              class="v-icon fas fa-arrow-left theme--light"
            />
             
            <span
              class="mx-3"
              style="line-height: 1.60rem;"
            >
              Return to template list
            </span>
          </div>
        </div>
      </button>
    </div>
     
    <div
      class="flex d-flex justify-space-between align-content-center my-3"
    >
      <div
        class="white--text headline"
      >
        
        New Document template
      
      </div>
    </div>
     
    <div
      class="flex mt-5"
    >
      <div
        class="layout row wrap"
      >
        <div
          class="flex xs12 sm12 md8 xl9"
        >
          <div
            class="ml-2 mt-2 v-card v-sheet theme--light"
          >
            <div
              class="v-card__title"
            >
              <h3
                class="grey--text"
              >
                
          Basic Information
        
              </h3>
               
              <!---->
            </div>
             
            <hr
              class="v-divider theme--light"
            />
             
            <div
              class="v-card__text"
            >
              <div>
                <div
                  class="flex xs12"
                >
                  <div
                    class="mb-2 text-uppercase grey--text font-weight-bold"
                  >
                    
              Title
            
                  </div>
                   
                  <div
                    class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <input
                            placeholder="Title"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="flex xs12"
                >
                  <div
                    class="mb-2 text-uppercase grey--text font-weight-bold"
                  >
                    
              Short description
            
                  </div>
                   
                  <div
                    class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <input
                            placeholder="Short description"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="flex xs12"
                >
                  <div
                    class="mb-2 text-uppercase grey--text font-weight-bold"
                  >
                    
              Content
            
                  </div>
                   
                  <div
                    class="editor"
                  >
                    <div
                      class="menubar"
                    >
                      <button
                        class="menubar__button"
                      >
                        
        H1
      
                      </button>
                       
                      <button
                        class="menubar__button"
                      >
                        
        H2
      
                      </button>
                       
                      <button
                        class="menubar__button"
                      >
                        
        H3
      
                      </button>
                       
                      <button
                        class="v-btn v-btn--icon v-btn--depressed theme--light"
                        type="button"
                      >
                        <div
                          class="v-btn__content"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            format_bold
                          </i>
                        </div>
                      </button>
                       
                      <button
                        class="v-btn v-btn--icon v-btn--depressed theme--light"
                        type="button"
                      >
                        <div
                          class="v-btn__content"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            format_italic
                          </i>
                        </div>
                      </button>
                       
                      <button
                        class="v-btn v-btn--icon v-btn--depressed theme--light"
                        type="button"
                      >
                        <div
                          class="v-btn__content"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            format_underlined
                          </i>
                        </div>
                      </button>
                    </div>
                     
                    <div
                      class="editor__content pa-3"
                    >
                      <div
                        class="ProseMirror"
                        contenteditable="true"
                        tabindex="0"
                      >
                        <p>
                          <br />
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-card__actions"
            >
              <button
                class="v-btn v-btn--depressed theme--light"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
          Cancel
        
                </div>
              </button>
               
              <button
                class="v-btn v-btn--depressed theme--light primary"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
          Submit
        
                </div>
              </button>
               
              <div
                class="spacer"
              />
               
              <button
                class="v-btn v-btn--depressed theme--light"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
          Delete
        
                </div>
              </button>
            </div>
          </div>
        </div>
         
        <div
          class="flex xs12 sm12 md4 xl3"
        >
          <div
            class="ml-2 mt-2 v-card v-sheet theme--light"
          >
            <div
              class="v-card__text"
            >
              <!---->
               
              <!---->
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-select theme--light"
                role="combobox"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-select__slot"
                    >
                      <div
                        class="v-select__selections"
                      >
                        <input
                          aria-readonly="false"
                          autocomplete="on"
                          readonly="readonly"
                          type="text"
                        />
                      </div>
                      <div
                        class="v-input__append-inner"
                      >
                        <div
                          class="v-input__icon v-input__icon--append"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            arrow_drop_down
                          </i>
                        </div>
                      </div>
                    </div>
                    <div
                      class="v-menu"
                    >
                      <div
                        class="v-menu__content theme--light "
                        style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                      >
                        <div
                          class="v-select-list v-card theme--light"
                        >
                          <div
                            class="v-list theme--light"
                            role="list"
                          >
                            <div
                              role="listitem"
                            >
                              <div
                                class="v-list__tile theme--light"
                              >
                                <div
                                  class="v-list__tile__content"
                                >
                                  <div
                                    class="v-list__tile__title"
                                  >
                                    No data available
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-card__actions"
            >
              <div
                class="layout text-xs-center row wrap"
              >
                <div
                  class="flex mb-2 xs12"
                >
                  <button
                    class="v-btn v-btn--block v-btn--depressed theme--light primary"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      
              Submit
            
                    </div>
                  </button>
                </div>
                 
                <div
                  class="flex mb-2 xs12"
                >
                  <button
                    class="v-btn v-btn--block v-btn--disabled v-btn--depressed theme--light"
                    disabled="disabled"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      
              Preview
            
                    </div>
                  </button>
                </div>
                 
                <div
                  class="flex xs12"
                >
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;

exports[`components/doc-templates/CreateUpdate Snapshots new 1`] = `
<div>
  <div
    class="layout px-4 pt4 column"
  >
    <div
      class="flex"
    >
      <button
        class="mx-0 v-btn v-btn--flat v-btn--small theme--light"
        type="button"
      >
        <div
          class="v-btn__content"
        >
          <div
            class="d-flex align-content-center white--text"
          >
            <i
              aria-hidden="true"
              class="v-icon fas fa-arrow-left theme--light"
            />
             
            <span
              class="mx-3"
              style="line-height: 1.60rem;"
            >
              Return to template list
            </span>
          </div>
        </div>
      </button>
    </div>
     
    <div
      class="flex d-flex justify-space-between align-content-center my-3"
    >
      <div
        class="white--text headline"
      >
        
        New Document template
      
      </div>
    </div>
     
    <div
      class="flex mt-5"
    >
      <div
        class="layout row wrap"
      >
        <div
          class="flex xs12 sm12 md8 xl9"
        >
          <div
            class="ml-2 mt-2 v-card v-sheet theme--light"
          >
            <div
              class="v-card__title"
            >
              <h3
                class="grey--text"
              >
                
          Basic Information
        
              </h3>
               
              <!---->
            </div>
             
            <hr
              class="v-divider theme--light"
            />
             
            <div
              class="v-card__text"
            >
              <div>
                <div
                  class="flex xs12"
                >
                  <div
                    class="mb-2 text-uppercase grey--text font-weight-bold"
                  >
                    
              Title
            
                  </div>
                   
                  <div
                    class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <input
                            placeholder="Title"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="flex xs12"
                >
                  <div
                    class="mb-2 text-uppercase grey--text font-weight-bold"
                  >
                    
              Short description
            
                  </div>
                   
                  <div
                    class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-text-field--placeholder theme--light"
                  >
                    <div
                      class="v-input__control"
                    >
                      <div
                        class="v-input__slot"
                      >
                        <div
                          class="v-text-field__slot"
                        >
                          <input
                            placeholder="Short description"
                            type="text"
                          />
                        </div>
                      </div>
                      <div
                        class="v-text-field__details"
                      >
                        <div
                          class="v-messages theme--light"
                        >
                          <div
                            class="v-messages__wrapper"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                 
                <div
                  class="flex xs12"
                >
                  <div
                    class="mb-2 text-uppercase grey--text font-weight-bold"
                  >
                    
              Content
            
                  </div>
                   
                  <div
                    class="editor"
                  >
                    <div
                      class="menubar"
                    >
                      <button
                        class="menubar__button"
                      >
                        
        H1
      
                      </button>
                       
                      <button
                        class="menubar__button"
                      >
                        
        H2
      
                      </button>
                       
                      <button
                        class="menubar__button"
                      >
                        
        H3
      
                      </button>
                       
                      <button
                        class="v-btn v-btn--icon v-btn--depressed theme--light"
                        type="button"
                      >
                        <div
                          class="v-btn__content"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            format_bold
                          </i>
                        </div>
                      </button>
                       
                      <button
                        class="v-btn v-btn--icon v-btn--depressed theme--light"
                        type="button"
                      >
                        <div
                          class="v-btn__content"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            format_italic
                          </i>
                        </div>
                      </button>
                       
                      <button
                        class="v-btn v-btn--icon v-btn--depressed theme--light"
                        type="button"
                      >
                        <div
                          class="v-btn__content"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            format_underlined
                          </i>
                        </div>
                      </button>
                    </div>
                     
                    <div
                      class="editor__content pa-3"
                    >
                      <div
                        class="ProseMirror"
                        contenteditable="true"
                        tabindex="0"
                      >
                        <p>
                          <br />
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-card__actions"
            >
              <button
                class="v-btn v-btn--depressed theme--light"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
          Cancel
        
                </div>
              </button>
               
              <button
                class="v-btn v-btn--depressed theme--light primary"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
          Submit
        
                </div>
              </button>
               
              <div
                class="spacer"
              />
               
              <button
                class="v-btn v-btn--depressed theme--light"
                type="button"
              >
                <div
                  class="v-btn__content"
                >
                  
          Delete
        
                </div>
              </button>
            </div>
          </div>
        </div>
         
        <div
          class="flex xs12 sm12 md4 xl3"
        >
          <div
            class="ml-2 mt-2 v-card v-sheet theme--light"
          >
            <div
              class="v-card__text"
            >
              <!---->
               
              <!---->
               
              <div
                class="v-input v-text-field v-text-field--single-line v-text-field--enclosed v-text-field--outline v-select theme--light"
                role="combobox"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                  >
                    <div
                      class="v-select__slot"
                    >
                      <div
                        class="v-select__selections"
                      >
                        <input
                          aria-readonly="false"
                          autocomplete="on"
                          readonly="readonly"
                          type="text"
                        />
                      </div>
                      <div
                        class="v-input__append-inner"
                      >
                        <div
                          class="v-input__icon v-input__icon--append"
                        >
                          <i
                            aria-hidden="true"
                            class="v-icon material-icons theme--light"
                          >
                            arrow_drop_down
                          </i>
                        </div>
                      </div>
                    </div>
                    <div
                      class="v-menu"
                    >
                      <div
                        class="v-menu__content theme--light "
                        style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                      >
                        <div
                          class="v-select-list v-card theme--light"
                        >
                          <div
                            class="v-list theme--light"
                            role="list"
                          >
                            <div
                              role="listitem"
                            >
                              <div
                                class="v-list__tile theme--light"
                              >
                                <div
                                  class="v-list__tile__content"
                                >
                                  <div
                                    class="v-list__tile__title"
                                  >
                                    No data available
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <div
                        class="v-messages__wrapper"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
             
            <div
              class="v-card__actions"
            >
              <div
                class="layout text-xs-center row wrap"
              >
                <div
                  class="flex mb-2 xs12"
                >
                  <button
                    class="v-btn v-btn--block v-btn--depressed theme--light primary"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      
              Submit
            
                    </div>
                  </button>
                </div>
                 
                <div
                  class="flex mb-2 xs12"
                >
                  <button
                    class="v-btn v-btn--block v-btn--disabled v-btn--depressed theme--light"
                    disabled="disabled"
                    type="button"
                  >
                    <div
                      class="v-btn__content"
                    >
                      
              Preview
            
                    </div>
                  </button>
                </div>
                 
                <div
                  class="flex xs12"
                >
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
   
  <!---->
</div>
`;
